pkg/cloud/services/compute/instance.go

```diff
@@ -99,25 +100,18 @@ func (s *Service) createInstanceImpl(eventObject runtime.Object, instanceSpec *I
 		Tags:             instanceSpec.Tags,
 		Metadata:         instanceSpec.Metadata,
 		ConfigDrive:      &instanceSpec.ConfigDrive,
+		BlockDevice:      blockDevices,
 	}
 
-	blockDevices, err := s.getBlockDevices(eventObject, instanceSpec, instanceSpec.ImageID, instanceCreateTimeout, retryInterval)
-	if err != nil {
-		return nil, err
-	}
-	if len(blockDevices) > 0 {
-		serverCreateOpts = bootfromvolume.CreateOptsExt{
+	server, err := s.getComputeClient().CreateServer(
```

Comment:
I am wondering whether you need a the Context.TODO() here in order to make this call happy 
so it's like not related to mock issue,but some param from reall call and test code consistent...

https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/2107/files#diff-b1233b605e340fb98df3c9fd0a05184538843bc5fb7ee2ce725d944c0868e8b8R267

Reply:
Matt seems to have found a solution to this issue, I am resolving this conversation. Please let me know if it was not the case!